### PR TITLE
[Examples] Update DGL Examples

### PR DESCRIPTION
**Pull request by yaox12**
1. Update the DGL example to make it work. 
2. Add multiple feature storage locations (`cpu|gpu|quiver|unified`) and CPU/GPU sampling support. Also attach a Readme for more details.
3. Modify the training script to have similar outputs and evaluation logic as the PyG version.
4. Remove `examples/dgl/ogbn_products_sage_dgl.py` because it's included in the current unified version.

## To Do
- [x] Single-GPU example
- [ ] Multi-GPU example (maybe in another PR)

### CODE REVIEW

**Comment by yaox12** [comment]
Hi maintainers, any advice on the single-GPU DGL example? If overall good, I will go ahead with the multi-GPU example.

**Comment by eedalong** [comment]
@yaox12  Hi， Great Job！We will discuss about this!

File: examples/dgl/ogbn_products_sage_quiver.py
Comment (ZenoTan) [inline examples/dgl/ogbn_products_sage_quiver.py:75]: Why batch size * 4 here?

File: examples/dgl/ogbn_products_sage_quiver.py
Comment (ZenoTan) [inline examples/dgl/ogbn_products_sage_quiver.py:78]: Ditto

File: examples/dgl/ogbn_products_sage_quiver.py
Comment (ZenoTan) [inline examples/dgl/ogbn_products_sage_quiver.py:232]: You could keep the original root or move it to args

**Comment by eedalong** [comment]
@yaox12 Single GPU example is great to me, please move on to multi-gpu examples! Looking forward to that!

File: examples/dgl/ogbn_products_sage_quiver.py
Comment (yaox12) [reply examples/dgl/ogbn_products_sage_quiver.py]: Fixed

File: examples/dgl/ogbn_products_sage_quiver.py
Comment (yaox12) [reply examples/dgl/ogbn_products_sage_quiver.py]: Fixed

File: examples/dgl/ogbn_products_sage_quiver.py
Comment (yaox12) [reply examples/dgl/ogbn_products_sage_quiver.py:75]: Use a larger batch size for evaluation, following the PyG version that uses a batch size of 1024 for training while 4096 for validation.

**Comment by luomai** [comment]
Thanks for the PR. I am thinking if we can make single-GPU DGL and multi-GPU DGL two separated PRs.

File: examples/dgl/README.md
Comment (eedalong) [inline examples/dgl/README.md:37]: Unified Tensor + CPU is much slower than Cache(21%) + CPU, I suspect this is related to host memory bus contention

**Comment by yaox12** [comment]
Sure, I'm supportive of putting the multi-GPU DGL examples in another PR. I'm a little busy this week and haven't got much time to finish it. Feel free to merge this PR if it looks good for you. Thanks.

File: examples/dgl/README.md
Comment (yaox12) [reply examples/dgl/README.md:37]: Agree. The transportation of sampled subgraphs from host to device may have competition with UnifiedTensor.

Comment (eedalong) [reply examples/dgl/README.md:37]: > 

Agree.

**Comment by eedalong** [comment]
> Sure, I'm supportive of putting the multi-GPU DGL examples in another PR. I'm a little busy this week and haven't got much time to finish it. Feel free to merge this PR if it looks good for you. Thanks.

Great!, We will merge this pr first

**Review by eedalong — APPROVED** [review]

**Comment by luomai** [comment]
Merged. Thanks a lot for the PR. @yaox12 :)

**Pull request merged**
